src/core/sequence_batch_scheduler.cc

```diff
@@ -1041,8 +1050,47 @@ DirectSequenceBatch::SchedulerThread(
               }
             }
 
+            earliest_enqueue_time_ns = std::min(
+                earliest_enqueue_time_ns, queue.front()->QueueStartNs());
+            ready_cnt++;
             max_seq_slot = seq_slot;
+          }
+        }
+
+        if (max_seq_slot != -1) {
```

Comment:
Can we also check pending_batch_delay and minimum_slot_util here and just skip the whole thing if either is zero.